hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BPOfferService.java

```diff
@@ -1163,14 +1105,60 @@ void notifyNamenodeDeletedBlockInt(
       // In case of un/successful block reports we have to inform the leader that
       // block reporting has finished for now.
       if (blkReportHander != null) {
-        getLeaderActor().blockReportCompleted(bpRegistration);
+        BPServiceActor leader =  getLeaderActor();
```

Comment:
shouldn't we retry if lead is null?

Reply:
The check is already there but it does not show here on github